packages/bot-web-ui/src/components/dashboard/dashboard-component/__tests__/dashboard-components.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import UserGuide from '../user-guide';
+import Sidebar from 'Components/dashboard/tutorial-tab/sidebar';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect:
+        () =>
+        <T,>(Component: T) =>
+            Component,
+}));
+
+jest.mock('@deriv/bot-skeleton', () => ({
+    Blockly: jest.fn(),
+}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Tabs: jest.fn(() => true),
+    performSelfExclusionCheck: jest.fn(),
+}));
+
+describe('<Dashboard />', () => {
+    it('user guide button click', () => {
+        const mocked_props = {
+            dashboard_strategies: [],
+            has_started_onboarding_tour: false,
+            setActiveTab: jest.fn(),
+            handleTabChange: jest.fn(),
+            active_tab: '0',
+            performSelfExclusionCheck: jest.fn(),
+            Blockly: jest.fn(),
+            dialog_options: {
+                title: 'string',
+                message: 'string',
+                ok_button_text: 'string',
+                cancel_button_text: 'string',
+            },
+            faq_search_value: 'test',
+            guide_list: [],
+            is_dialog_open: true,
+            onOkButtonClick: jest.fn(),
+            setHasTourEnded: jest.fn(),
+            setOnBoardTourRunState: jest.fn(),
+            setTourActiv: jest.fn(),
+            setTourDialogVisibility: jest.fn(),
+            showVideoDialog: jest.fn(),
+        };
+        // need to use the dashboard component here but commented because it dependent on
+        // self exclusion check so kept this component else we can use dashboard throughout
+        // using indivisual components for now
+        //render(<Dashboard {...mocked_props} />);
+        render(<UserGuide {...mocked_props} />);
+        const use_guide_button = screen.getByTestId('btn-use-guide');
+        userEvent.click(use_guide_button);
```

Comment:
A so-called "assert" section is missing here. Any test case which is supposed to consist of arrange, act and assert sections. Please add assertion to this test case where you need to test the rendering of the component using "expect(" 
After you "render(" your component, you can test if it's rendered correctly using assertions, and after that in the same test case you can click the button with "userEvent.click("  and test if the component is rendered correctly after the click by checking some elements that were supposed to change, or callbacks that were supposed to be called with certain params. And then in the second test case you can check if the component is rendered properly when it receives another set of props.👍🏻 It's unit testing so first of all we need to test the component's functionality from user perspective and in an isolated (test) environment that Jest creates when we run the test. So if the behavior and rendering will change based on props, we have to add test cases that cover these cases.

Reply:
@maryia-deriv i have divided the components now into user guide which is a unit test component and a dashboard component which is a integration test component. 
Let me know if i need to do changes. 🙇

Reply:
oh, i can see you test other components here. We should not render(<UserGuide {...mocked_props} />) or render(<Sidebar {...mocked_props} />) in the tests for Dashboard. We should only test and render <DashboardComponent /> in this test file which should be renamed to dashboard-component.spec.tsx🙂